Mathlib/Data/Int/Card.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Data.Int.Interval
+import Mathlib.Data.Int.ModEq
+import Mathlib.Data.Nat.Count
+import Mathlib.Data.Rat.Floor
+
+/-!
+# Number of elements in an integer/natural number interval congruent to a given number
+
+Pass
+-/
+
+
+open Finset Int
+
+/-- There are `⌈b / r⌉ - ⌈a / r⌉` multiples of `r` in `[a, b)`, if `a ≤ b`. -/
+theorem Int.Ico_filter_dvd_card (a b : ℤ) {r : ℤ} (hr : 0 < r) :
+    ((Ico a b).filter (r ∣ ·)).card = max (⌈b / (r : ℚ)⌉ - ⌈a / (r : ℚ)⌉) 0 := by
```

Comment:
I think you should actually prove
```suggestion
    (Ico a b).filter (r ∣ ·) = (Ico ⌈b / (r : ℚ)⌉ ⌈a / (r : ℚ)⌉).map (.mul_right) := by
```
Or similar

Reply:
That was in fact how my original proof went (_separately_ establishing equality of sets), but @b-mehta (when I presented the declaration for golfing on Xena) thought that _integrating_ the equivalence proof into the cardinality theorem through `card_congr` was better. I agreed with him.

Reply:
…never mind, your suggestion turns out to be even shorter.